api/doc/openapi.yaml

```diff
@@ -272,6 +272,44 @@ paths:
           $ref: '#/components/responses/415'
         "500":
           $ref: '#/components/responses/500'
+  /transactions/simulate:
+    post:
+      summary: Simulate transaction
+      operationId: simulate_transaction
+      description: |
+        **Submit transaction for simulation result using JSON **
+
+          * Send [POST /transactions/signing_message](#operation/create-signing-message) to create transaction signing message.
+          * Sign the transaction signing message and create transaction signature.
```

Comment:
Is it a security concern requiring real signatures on the transaction to be dry-runed?

Reply:
I agree, this API shouldn't take in signed transactions if at all possible. These transactions should assume that the user is willing to pay whatever gas required to test -- in theory this should probably contain only. If we support signed, then a user could get themselves into a lot of trouble.

[sender, transaction payload]

for ease, maybe we assume RawTransaction for bcs and signing for REST?

Reply:
great point!

Reply:
But giving AptosVM the ability to create fake `SignedTransaction` sounds like a can of worm I'd rather not open. Might get abused down the road one day.

One possibility that protects the user is explicitly adding a field "is_simulation" to `UserTransactionRequest`, and at the same time probably setting gas_unit_price to 0. And the mempool should reject any txn whose `is_simulation` is true.